src/interop/op-supervisor/op_supervisor_launcher.star

```diff
@@ -30,56 +30,82 @@ DATA_DIR = "/etc/op-supervisor"
 DEPENDENCY_SET_FILE_NAME = "dependency_set.json"
 
 
-def create_dependency_set(chains):
+def create_dependency_set(l2s):
     result = {
         "dependencies": {
-            str(chain.network_params.network_id): {
-                "chainIndex": str(chain.network_params.network_id),
+            str(l2.network_id): {
+                "chainIndex": str(l2.network_id),
                 "activationTime": 0,
                 "historyMinTime": 0,
             }
-            for chain in chains
+            for l2 in l2s
         }
     }
     return result
 
 
 def launch(
     plan,
+    interop_set,
     l1_config_env_vars,
-    chains,
     l2s,
     jwt_file,
-    supervisor_params,
     observability_helper,
 ):
-    dependency_set_json = supervisor_params.dependency_set
-    if not dependency_set_json:
-        dependency_set = create_dependency_set(chains)
-        dependency_set_json = json.encode(dependency_set)
+    # First we check that the supervisor is enabled for this interop set
+    if not interop_set.enabled:
+        plan.print(
+            "op-supervisor is not enabled for interop set {}, skipping launch".format(
+                interop_set.name
+            )
+        )
+        return None
+
+    # Then we check that we have some participants
+    if len(interop_set.participants) == 0:
+        plan.print(
+            "op-supervisor has no participants for interop set {}, skipping launch".format(
+                interop_set.name
+            )
+        )
+        return None
+
+    # Now we filter out the participating L2s
+    interop_set_l2s = [l2 for l2 in l2s if l2.network_id in interop_set.participants]
+
+    # Now we create dependency set if none was provided
+    dependency_set_json = interop_set.supervisor_params.dependency_set or json.encode(
+        create_dependency_set(interop_set_l2s)
+    )
 
+    # And write it to an artifact
     dependency_set_artifact = utils.write_to_file(
```

Comment:
shouldn't this be adjusted as well so we one artifact per depset ?
I might be missing something, but it looks like we might be overwriting the same artifact here, which would presumably make all supervisors have the same config.

Reply:
True dat